### DISCUSSION TITLE
Building application on Vercel gives a Webpack error

### DISCUSSION THREAD

**Discussion by yanickvanbarneveld** [vercel/next.js#27372] [Help]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14.15.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Vercel

### Describe the Bug

When I am trying to build my application locally everything works as expected. When I am building my application on Vercel I retrieve the following error in the console:

`
10:02:44.120 | <w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules\|/vercel/path0/node_modules/next/dist/compiled/css-loader/cjs.js??ruleSet[1].rules[2].oneOf[3].use[1]!/vercel/path0/node_modules/next/dist/compiled/postcss-loader/cjs.js??ruleSet[1].rules[2].oneOf[3].use[2]!/vercel/path0/node_modules/next/dist/compiled/resolve-url-loader/index.js??ruleSet[1].rules[2].oneOf[3].use[3]!/vercel/path0/node_modules/next/dist/compiled/sass-loader/cjs.js??ruleSet[1].rules[2].oneOf[3].use[4]!/vercel/path0/src/components/Prepr/PreprImage/PreprImage.module.scss': No serializer registered for SassError
-- | --
10:02:44.120 | <w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> webpack/lib/ModuleBuildError -> SassError
10:02:45.238 | Failed to compile.
10:02:45.239 | ./src/components/Component/Council/Council.module.scss.webpack[javascript/auto]!=!./node_modules/next/dist/compiled/css-loader/cjs.js??ruleSet[1].rules[2].oneOf[3].use[1]!./node_modules/next/dist/compiled/postcss-loader/cjs.js??ruleSet[1].rules[2].oneOf[3].use[2]!./node_modules/next/dist/compiled/resolve-url-loader/index.js??ruleSet[1].rules[2].oneOf[3].use[3]!./node_modules/next/dist/compiled/sass-loader/cjs.js??ruleSet[1].rules[2].oneOf[3].use[4]!./src/components/Component/Council/Council.module.scss
10:02:45.239 | SassError: Can't find stylesheet to import.
10:02:45.239 | ╷
10:02:45.239 | 1 │ @import '@Assets/styles/abstracts';
10:02:45.239 | │         ^^^^^^^^^^^^^^^^^^^^^^^^^^
10:02:45.239 | ╵
10:02:45.239 | src/components/Component/Council/Council.module.scss 1:9  root stylesheet
10:02:45.239 | > Build error occurred
10:02:45.240 | Error: > Build failed because of webpack errors
10:02:45.240 | at /vercel/path0/node_modules/next/dist/build/index.js:15:924
10:02:45.241 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:6:584)
10:02:45.284 | error Command failed with exit code 1.
10:02:45.284 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
10:02:45.303 | Error: Command "yarn run build" exited with 1
`

My next.config.js looks like this:
`
const withPlugins = require('next-compose-plugins');
const { withSentryConfig } = require('@sentry/nextjs');
const withNextTranslate = require('next-translate');
const withBundleAnalyzer = require('@next/bundle-analyzer')({
    enabled: process.env.ANALYZE === 'true'
});

const nextConfiguration = {
    images: {
        domains: ['via.placeholder.com', '6scjpsqwjwm7.b-cdn.net']
    }
};

module.exports = withPlugins(
    [withSentryConfig, withNextTranslate, withBundleAnalyzer],
    nextConfiguration
);
`

### Expected Behavior

Build the application successfully without any errors.

### To Reproduce

-

**Comment by ijjk** [comment]
Hi, can you provide a repo with a minimal reproduction to allow investigation? Are you not seeing the same error when building locally with `next build`?

**Comment by jschuur** [comment]
This sounds similar to an issue I have just started experiencing today. Yesterday, I restructured part of my app (creating a subdirectory structure for my components, and now I get a build error when I push to git and let Vercel do the deployment, but it works fine both locally and even if I use `vercel --prod` to deploy directly from my machine:

```
15:19:06.001 | Failed to compile. 
15:19:06.001 | ModuleNotFoundError: Module not found: Error: Can't resolve '../components/Admin/AddChannelForm' in '/vercel/path0/pages'
15:19:06.003 | > Build error occurred
15:19:06.003 | Error: > Build failed because of webpack errors
15:19:06.003 | at /vercel/path0/node_modules/next/dist/build/index.js:15:924
15:19:06.003 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:6:584)
```

I don't have a minimal reproduction example yet, but my project can be viewed [here](https://github.com/jschuur/streamers.dev)

Initially, I thought my error came from my attempt at getting absolute import aliases working, but even when I rolled that back, I still had the same error with my new directory structure.

In my example, I have pages and components directories in the root, and it's failing to import `'../components/Admin/AddChannelForm'` in `pages/admin.js`.

**Comment by ijjk** [comment]
Hi, does the casing match what you have checked into your project? Your local fs can be case-insensitive which is why it works there but the fs on Vercel is case-sensitive.

**Comment by jschuur** [comment]
> Hi, does the casing match what you have checked into your project? Your local fs can be case-insensitive which is why it works there but the fs on Vercel is case-sensitive.

In my case I'm on macOS, so the fs is case sensitive and the casing matches locally too. My component file has also not changed casing since I reorganised and it used to work. I just added the Admin folder under components (among others).

BTW, I also started a support ticket on this earlier (#00036536), but don't have anything yet where I can reply there to provide the extra info on how it does work if I deploy via the local CLI.

**Comment by ijjk** [comment]
@jschuur macOS is not case sensitive unless customized, see related discussion [here](https://discussions.apple.com/thread/251191099#:~:text=No.,have%20two%20files%20named%20File) you can check the casing is correct in git by doing `git mv ./path/to/components/admin/addChannelForm ./path/to/components/Admin/AddChannelForm'` see related discussion [here](https://stackoverflow.com/questions/17683458/how-do-i-commit-case-sensitive-only-filename-changes-in-git)

**Comment by jschuur** [comment]
Oh yeah, now I see the issue, thanks! Git didn't update the file (by
design) because I had just changed the directory capitalisation.

Thanks for the pointer. Now it builds fine on Vercel.

On Wed, Jul 21, 2021 at 4:30 PM JJ Kasper ***@***.***> wrote:

> @jschuur <https://github.com/jschuur> macOS is not case sensitive unless
> customized, see related discussion here
> <https://discussions.apple.com/thread/251191099#:~:text=No.,have%20two%20files%20named%20File>
> you can check the casing is correct in git by doing git mv
> ./path/to/components/admin/addChannelForm
> ./path/to/components/Admin/AddChannelForm' see related discussion here
> <https://stackoverflow.com/questions/17683458/how-do-i-commit-case-sensitive-only-filename-changes-in-git>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/next.js/issues/27327#issuecomment-884282379>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAWSKNNH5DNMYWT4RYSLSDTY3RZPANCNFSM5AVI457Q>
> .
>


-- 
Twitter <http://twitter.com/joostschuur>

**Comment by ijjk** [comment]
@yanickvanbarneveld can you confirm your issue is resolved by doing the same?

**Comment by tcetin** [comment]
I have the same problem. I just upgraded to 13.4.5, and it gives this error : No serializer registered for SassError
